doc/api.md

```diff
@@ -452,20 +460,408 @@ Note that, even creating a group with `autoprovision=true` and `explicitAttrs=tr
+the Context Broker.
+
+E.g.: if a device with the following provisioning information is provisioned in the IoT Agent:
+
+```json
+{
+   "name":"location",
+   "type":"geo:point",
+   "expression": "longitude+', '+latitude"
+},
+{
+   "name":"fillingLevel",
+   "type":"Number",
+   "expression": "level / 100",
+},
+```
+
+[Interactive expression `longitude+', '+latitude`][3]
+
+[Interactive expression `level / 100`][4]
+
+and a measurement with the following values arrive to the IoT Agent:
+
+```text
+latitude: 1.9
+level: 85.3
+```
+
+The only expression rule that will be executed will be that of the `fillingLevel` attribute. It will produce the value
+`0.853` that will be sent to the Context Broker.
+
+Note that expressions are only applied if the attribute name (as received by the IoT Agent in the southbound interface)
+matches the expression variable. Otherwise, the southbound value is used directly. Let's illustrate with the following
+example:
+
+```json
+"consumption": {
+   "type": "String",
+   "value": "spaces | trim"
+}
+```
+
+-   Case 1: the following measure is received at the southbound interface:
+
+```text
+consumption: "0.44"
+```
+
+As `spaces` attribute is not included, then the expression is not applied and the `consumption` measure value is
+directly used, so the following is sent to CB:
+
+```json
+"consumption": {
+   "type": "String",
+   "value": "0.44"
+}
+```
+
+-   Case 2: the following measure is received at the southbound interface:
+
+```text
+consumption: "0.44"
+spaces: "  foobar  "
+```
+
+As `spaces` attribute is included, then the expression is evaluated, so overriding the 0.44 value and sending the
+following to CB:
+
+```json
+"consumption": {
+    "type": "String",
+    "value": "foobar"
+}
+```
+
+[Interactive expression `spaces | trim`][5]
+
+### Multientity measurement transformation support (`object_id`)
+
+To allow support for measurement transformation in combination with multi entity plugin, where the same attribute is
+generated for different entities out of different incoming attribute values (i.e. `object_id`), we introduced support
+for `object_id` in the expression context.
+
+For example, the following device:
+
+```json
+"WeatherStation": {
+    "commands": [],
+    "type": "WeatherStation",
+    "lazy": [],
+    "active": [
+        {
+            "object_id": "v1",
+            "name": "vol",
+            "expression" : "v1*100",
+            "type": "Number",
+            "entity_name": "WeatherStation1"
+        },
+        {
+            "object_id": "v2",
+            "name": "vol",
+            "expression" : "v2*100",
+            "type": "Number",
+            "entity_name": "WeatherStation2"
+        },
+        {
+            "object_id": "v",
+            "name": "vol",
+            "expression" : "v*100",
+            "type": "Number"
+        }
+    ]
+}
+```
+
+When receiving the attributes `v`, `v1` and `v2` in a payload from a message received in the southbound:
```

Comment:
In fact, what are received are *measure* not *attributes*. Attributes is the conversion of the measure in the southbound to the Context Broker context in the northbound.

Maybe better:

```suggestion
When receiving the measures `v`, `v1` and `v2` in a payload from a message received in the southbound:
```

Reply:
I Agree, better to use measure term